src/store/auth.js

```diff
@@ -0,0 +1,43 @@
+import api from '../plugins/axios';
+import { saveToken } from '../common/jwt.service';
+
+const state = () => ({
+  curUser: {},
+  isAuthenticated: false
+});
+
+const getters = {
+  curUser: state => state.curUser,
+  isAuth: state => state.isAuthenticated
+};
+
+const mutations = {
+  setAuthState: (state, isAuth) => (state.isAuthenticated = isAuth)
+};
+
+const actions = {
+  // sign in user and save token
+  signin({ commit }, user) {
+    api
+      .post('/users/signin', user)
+      .then(res => {
+        console.log(res.data.message);
+        saveToken(res.data.token);
+        commit('setAuthState', true);
+      })
+      .catch(error => console.log('Sign in ERROR --- ' + error));
+  },
+  // SIGN UP user and save token
+  signup({ commit }, user) {
+    api
+      .post('/users/signin', user)
+      .then(res => {
+        console.log(res.data.message);
```

Comment:
we should be check response data be for do any action to make sure properties of object exist
if (res) {
}